README.md

```diff
@@ -42,6 +42,56 @@ If you have both this extension and SmartBlocks installed, there is a `<%GOOGLEC
 - `<%GOOGLECALENDAR:today,tomorrow%>`
 - `<%GOOGLECALENDAR:tomorrow%>`
 
+### Developer API
+
+The RoamJS Google Calendar integration offers an API to allow other extensions and custom scripts to fetch Google Calendar events directly within Roam Research. This functionality is exposed through the global `window.roamjs.extension.google` object.
+
+#### fetchGoogleCalendar
+
+**Description**
+
+Fetches events from the user's Google Calendar(s) for a specified date range.
+
+**Syntax**
+
+```javascript
+window.roamjs.extension.google.fetchGoogleCalendar(
+  startDatePageTitle,
+  endDatePageTitle
+);
+```
+
+**Parameters**
+
+- `startDatePageTitle` (optional) - A string representing the title of the page that corresponds to the start date of the desired events. The date should be in a format recognized by Roam's `window.roamAlphaAPI.util.pageTitleToDate` utility. If not provided, the current focused page title is used as the start date.
+- `endDatePageTitle` (optional) - A string representing the title of the page that corresponds to the end date of the desired events. The date should be in a format recognized by Roam's `window.roamAlphaAPI.util.pageTitleToDate` utility. If not provided, the start date is used as the end date.
+
+**Returns**
+
+A Promise that resolves to an array of `InputTextNode[]`. Each InputTextNode represents an event fetched from Google Calendar, formatted according to the user's settings. The structure of an InputTextNode is as follows:
+
+```typescript
+interface InputTextNode {
+  text: string;
+  children?: InputTextNode[];
+}
+```
+
+**Example**
+
+```javascript
+window.roamjs.extension.google
+  .fetchGoogleCalendar("January 1st, 2024", "January 31st, 2024")
```

Comment:
I'd make this an object `{ startDatePageTitle, endDatePageTitle }`. Eventually, there will be a request for start/end uids, start/end nlps, etc